src/containers/pages/InterventionPlan/NewPlan/General/index.tsx

```diff
@@ -1,15 +1,54 @@
 import React, { useState } from 'react';
 import { Helmet } from 'react-helmet';
 import { Col, Row } from 'reactstrap';
-import PlanForm, { defaultInitialValues } from '../../../../../components/forms/PlanForm';
+import PlanForm, {
+  defaultInitialValues,
+  PlanFormProps,
+} from '../../../../../components/forms/PlanForm';
+import { getFormActivities, IRSActivities } from '../../../../../components/forms/PlanForm/helpers';
 import HeaderBreadcrumb from '../../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
-import { CREATE_NEW_PLAN, HOME, PLANS } from '../../../../../configs/lang';
-import { HOME_URL, NEW_PLAN_URL, PLAN_LIST_URL } from '../../../../../constants';
+import { COUNTRY, CREATE_NEW_PLAN, HOME, PLANS } from '../../../../../configs/lang';
+import {
+  ASSIGN_JURISDICTIONS_URL,
+  HOME_URL,
+  NEW_PLAN_URL,
+  PLAN_LIST_URL,
+} from '../../../../../constants';
 import { InterventionType } from '../../../../../store/ducks/plans';
 import { JurisdictionDetails } from './JurisdictionDetails';
 
+/** expose props that would enable one to customize the underlying planForm props */
+interface BaseNewPlanProps {
+  extraPlanFormProps: Partial<PlanFormProps>;
+}
+
+/** the default props */
+export const defaultBasePlanProps = {
+  extraPlanFormProps: { redirectAfterAction: PLAN_LIST_URL },
+};
+
+/** dynamically supply props during runtime when the interventionType changes
+ * ps: the planForm passed in the props take the highest precedence
+ */
+const planFormPropsLookUp = {
+  [InterventionType.IRS]: {
+    allowMoreJurisdictions: false,
+    cascadingSelect: false,
+    initialValues: {
+      ...defaultInitialValues,
+      activities: getFormActivities(IRSActivities),
+      interventionType: InterventionType.IRS,
+    },
+    jurisdictionLabel: COUNTRY,
+    redirectAfterAction: ASSIGN_JURISDICTIONS_URL,
+  },
+  [InterventionType.FI]: {},
+  [InterventionType.MDA]: {},
+  [InterventionType.MDAPoint]: {},
+};
```

Comment:
Do we expect to always have different props for different plan types?

Reply:
at the moment , no, added those so that I could have a unified of interface when spreading the props inside the component, i.e. did not want to keep checking if a value is undefined before applying the lookup'd props